gui/datumaro_gui/components/multiple/streamlit_app.py

```diff
@@ -18,13 +18,15 @@
     save_dataset,
 )
 from datumaro_gui.utils.drawing.css import box_style, custom_css
+from datumaro_gui.utils.page import check_image_backend
 from datumaro_gui.utils.readme import github_pypi_desc
 from streamlit import session_state as state
 
 from . import tabs
 
 
 def main():
```

Comment:
```suggestion
# Copy context vars somewhere close to initialization
from contextvars import copy_context
st.session_state["context_vars"] = copy_context()

...

from functools import wraps

def update_context_vars(func):
    @wraps(func):
    def wrapped(*args, **kwargs):
        if ctx := st.session_state.get("context_vars", None):
            for val, value in ctx.items():
                val.set(value)
        else:
            msg = "You should copy context variables to streamlit session_state first."
            logger.error(msg)
        return func(*args, **kwargs)
        
    return wrapped

@update_context_vars
def main():
```
Revise it to cover all context variables and consider to make it as a function decorator for a cleaner code pattern.

Reply:
I will cover this in following PR.